### PR TITLE
feat: option to generate code with strict types

### PR DESCRIPTION
**Pull request by MrMeshok**
PHP, by default, will coerce values of the wrong type into the expected scalar type declaration if possible. This leads to unexpected behavior. For example, losing decimal digits when casting float to int
```PHP
readonly class IntDto
{
    public function __construct(
        public int $var,
    ) {}
}

$autoMapper = \AutoMapper\AutoMapper::create();
$result = $autoMapper->map(['var' => 1.1111], IntDto::class);
$result->var; // var is 1
```
This PR adds configuration to add `declare(strict_types=1)` to generated code when mapping, so instead of unexpected behavior, you get TypeError.

If everything is ok, I will update the docs, but I am not very fluent with english, so maybe someone else could write a more understandable description.

### CODE REVIEW

**Comment by joelwurtz** [comment]
:+1: for this thanks, can you add the possibility to configure this through the `Mapper` attribute also ?

Also wondering if declare strict should be true by default ? poke @Korbeil @nikophil ?

If you could update the doc and add a note in the changelog it would be nice, if you are not comfortable doing so we can do it.

**Comment by Korbeil** [comment]
I'm totally for this option and I think we should put it by default as @joelwurtz stated.
Same as he said, we would require a configuration on the `Mapper` attribute :pray:

**Comment by nikophil** [comment]
won't we face some BC break if this becomes a default thing?

**Comment by MrMeshok** [comment]
Made it configurable from `Mapper` and enabled by default. I'll wait for the decision about the default value and force push if it needs to be changed. I will leave the changelog and docs for you

File: src/Metadata/GeneratorMetadata.php [hunk 1]
```diff
@@ -24,7 +24,8 @@ public function __construct(
         public readonly array $propertiesMetadata,
         public readonly bool $checkAttributes = true,
         public readonly ConstructorStrategy $constructorStrategy = ConstructorStrategy::AUTO,
-        public bool $allowReadOnlyTargetToPopulate = false,
+        public readonly bool $allowReadOnlyTargetToPopulate = false,
```
Comment (MrMeshok) [inline src/Metadata/GeneratorMetadata.php:27]: Seems like readonly was missed here

**Comment by Korbeil** [comment]
> won't we face some BC break if this becomes a default thing?

This is right, we should put it disabled by default and we will put it enabled by default on next major ~
Could you make the change @MrMeshok ? :pray:

**Comment by MrMeshok** [comment]
@Korbeil Done👍

**Review by Korbeil — APPROVED** [review]
Looks good to me :ok_hand:

Could you add a note about your feature into the CHANGELOG please ?

**Comment by MrMeshok** [comment]
Added note to CHANGELOG and combined everything in 1 commit

**Comment by Korbeil** [comment]
Thanks for the feature @MrMeshok ! :pray:

**Pull request merged**